kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandlerTest.java

```diff
@@ -379,11 +409,37 @@ void expectedFlow(boolean sslConfigured,
         var handler = handler(filter, dp, endpointBinding);
         initialiseInboundChannel(handler);
 
+        if (clientAuthConfigured) {
+            try {
+                when(sslSession.getPeerPrincipal()).thenReturn(mockPrincipal);
+            }
+            catch (SSLPeerUnverifiedException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
I don't think the try catch is useful here, just mark `expectedFlow` as throwing `SSLPeerUnverifiedException`